### PR TITLE
Fixes multiple bugs found by test team

### LINKED ISSUE #86: Salt execution module junos.cli with invalid dest="" throws exception FileNotFoundError: [Errno 2] No such file or directory
### Description of Issue/Question
salt execution module junos.cli with invalid dest="" throws exception FileNotFoundError: [Errno 2] No such file or directory..
 Exception not handled ..

### Logs

root@testbase:/srv/salt# salt "vmx1" junos.cli "show version" dev_timeout=60 dest=""
[WARNING ] /usr/local/lib/python3.4/dist-packages/salt/transport/ipc.py:287: DeprecationWarning: encoding is deprecated, Use raw=False instead.
  self.unpacker = msgpack.Unpacker(encoding=encoding)

[WARNING ] /usr/local/lib/python3.4/dist-packages/salt/payload.py:149: DeprecationWarning: encoding is deprecated, Use raw=False instead.
  ret = msgpack.loads(msg, use_list=True, ext_hook=ext_type_decoder, encoding=encoding)

vmx1:
    The minion function caused an exception: Traceback (most recent call last):
      File "/usr/local/lib/python3.4/dist-packages/salt/minion.py", line 1660, in _thread_return
        return_data = minion_instance.executors[fname](opts, data, func, args, kwargs)
      File "/usr/local/lib/python3.4/dist-packages/salt/executors/direct_call.py", line 12, in execute
        return func(*args, **kwargs)
      File "/usr/local/lib/python3.4/dist-packages/salt/modules/junos.py", line 91, in wrapper
        result = function(*args, **kwargs)
      File "/usr/local/lib/python3.4/dist-packages/salt/modules/junos.py", line 689, in cli
        with salt.utils.files.fopen(op['dest'], 'w') as fp:
      File "/usr/local/lib/python3.4/dist-packages/salt/utils/files.py", line 399, in fopen
        f_handle = open(*args, **kwargs)  # pylint: disable=resource-leakage
    FileNotFoundError: [Errno 2] No such file or directory: ''
ERROR: Minions returned with non-zero exit code

### PR DESCRIPTION
**Pull request by rsmekala**
Fixes #86. 

Internal tracking PR: [1427511](https://gnats.juniper.net/web/default/1427511#description_tab)

### CODE REVIEW

**Comment by rsmekala** [comment]
Fixes #88.

Internal Tracking PR: [1427861](https://gnats.juniper.net/web/default/1427861-1#description_tab)

**Comment by rsmekala** [comment]
Fixes #89 

Internal Tracking PR: [1428147](https://gnats.juniper.net/web/default/1428147-1#audit_tab)

**Comment by rsmekala** [comment]
Fixes #90 

Internal Tracking PR: [1428175](https://gnats.juniper.net/web/default/1428175-1#description_tab)

New Behavior:

```
➜ salt 'vmx' junos.rpc get-config format=xml dest=/tmp/cli.out filter='<configuration>'

vmx:
    ----------
    message:
        Invalid filter XMLSyntaxError('Premature end of data in tag configuration line 1, line 1, column 16')
    out:
        False
```

**Comment by rsmekala** [comment]
Fixes #91

File: salt/engines/junos_syslog.py [hunk 1]
```diff
@@ -27,29 +27,27 @@
...
-
-The default topic title is ``jnpr/syslog/hostname/event``.
+ - jnpr/syslog/1.1.1.1/mgd/UI_COMMIT_COMPLETED
+ - jnpr/syslog/sshd/7
+The default topic title is 'jnpr/syslog/hostname/event'.
 
-One can choose the type of data they want from the event bus. For example, if
-one wants only events pertaining to a particular daemon, this can be specified
-in the configuration file:
+The user can choose the type of data he/she wants of the event bus.
```
Comment (vnitinv) [inline salt/engines/junos_syslog.py]: `he/she` change it to `they`

File: salt/engines/junos_syslog.py [hunk 2]
```diff
@@ -27,29 +27,27 @@
...
+Like, if one wants only events pertaining to a particular daemon, he/she can
```
Comment (vnitinv) [inline salt/engines/junos_syslog.py]: same here

**Review by vnitinv — APPROVED** [review]
do the requested changes please

File: salt/engines/junos_syslog.py (hunk 2 above)
Comment (rsmekala) [reply salt/engines/junos_syslog.py]: Done

File: salt/engines/junos_syslog.py (hunk 1 above)
Comment (rsmekala) [reply salt/engines/junos_syslog.py]: Done

**Pull request merged**